modules/git/repo_cleanup.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package git
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"code.gitea.io/gitea/modules/log"
+)
+
+const threshold = -1 * time.Hour
+
+func ForciblyUnlockRepository(ctx context.Context, repoPath string) error {
+	return cleanLocksIfNeeded(repoPath, time.Now())
+}
+
+func ForciblyUnlockRepositoryIfNeeded(ctx context.Context, repoPath string) error {
+	lockThreshold := time.Now().Add(threshold)
+	return cleanLocksIfNeeded(repoPath, lockThreshold)
+}
+
+func cleanLocksIfNeeded(repoPath string, threshold time.Time) error {
+	if repoPath == "" {
+		return nil
+	} else {
+		log.Trace("Checking if repository %s is locked [lock threshold is %s]", repoPath, threshold)
+		return filepath.Walk(repoPath, func(filePath string, fileInfo os.FileInfo, err error) error {
```

Comment:
I don't think Walk is necessary since the lock files have a fixed relative path.